internal/resource/config/sp/sp_authentication_policy_contract_mapping_resource.go

```diff
@@ -0,0 +1,285 @@
+		}
+	}
+	// Log response JSON
+	_, responseErr := apiReadSpAuthenticationPolicyContractMappingResource.MarshalJSON()
+	if responseErr != nil {
+		diags.AddError("There was an issue retrieving the response of the SpAuthenticationPolicyContractMappingResource: %s", responseErr.Error())
+	}
+
+	// Read the response into the state
+	diags = readSpAuthenticationPolicyContractMappingResourceResponse(ctx, apiReadSpAuthenticationPolicyContractMappingResource, &state)
+	resp.Diagnostics.Append(diags...)
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *spAuthenticationPolicyContractMappingResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var plan spAuthenticationPolicyContractMappingResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	attributeContractFulfillment := &map[string]client.AttributeFulfillmentValue{}
+	attributeContractFulfillmentErr := json.Unmarshal([]byte(internaljson.FromValue(plan.AttributeContractFulfillment, false)), attributeContractFulfillment)
+	if attributeContractFulfillmentErr != nil {
+		resp.Diagnostics.AddError("Failed to build attribute contract fulfillment request object:", attributeContractFulfillmentErr.Error())
+		return
+	}
+	updateSpAuthenticationPolicyContractMappingResource := r.apiClient.SpAuthenticationPolicyContractMappingsAPI.UpdateApcToSpAdapterMappingById(config.ProviderBasicAuthContext(ctx, r.providerConfig), plan.Id.ValueString())
+	createUpdateRequest := client.NewApcToSpAdapterMapping(*attributeContractFulfillment, plan.SourceId.ValueString(), plan.TargetId.ValueString())
+	err := addOptionalSpAuthenticationPolicyContractMappingResourceFields(ctx, createUpdateRequest, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for SpAuthenticationPolicyContractMappingResource", err.Error())
+		return
+	}
+	_, requestErr := createUpdateRequest.MarshalJSON()
+	if requestErr != nil {
+		diags.AddError("There was an issue retrieving the request of the SpAuthenticationPolicyContractMappingResource: %s", requestErr.Error())
+	}
+	updateSpAuthenticationPolicyContractMappingResource = updateSpAuthenticationPolicyContractMappingResource.Body(*createUpdateRequest)
+	updateSpAuthenticationPolicyContractMappingResourceResponse, httpResp, err := r.apiClient.SpAuthenticationPolicyContractMappingsAPI.UpdateApcToSpAdapterMappingByIdExecute(updateSpAuthenticationPolicyContractMappingResource)
+	if err != nil && (httpResp == nil || httpResp.StatusCode != 404) {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while updating SpAuthenticationPolicyContractMappingResource", err, httpResp)
+		return
+	}
+	// Log response JSON
```

Comment:
I'm not sure there's any reason to be running these MarshalJSON calls anymore now that we aren't printing out the body contents.